src/modules/settings/components/settings/PinnedHighlights.jsx

```diff
@@ -1,22 +1,35 @@
 import React from 'react';
 import Panel from 'rsuite/lib/Panel/index.js';
 import Toggle from 'rsuite/lib/Toggle/index.js';
+import Input from 'rsuite/lib/Input/index.js';
 import {registerComponent, useStorageState} from '../Store.jsx';
 import {CategoryTypes, SettingIds} from '../../../../constants.js';
 import styles from '../../styles/header.module.css';
 
-function PinnedHighlights() {
+const PinnedHighlights = () => {
   const [value, setValue] = useStorageState(SettingIds.PINNED_HIGHLIGHTS);
+  const [pinnedHighlights, setPinnedHighlights] = useStorageState(SettingIds.NUMBER_OF_PINNED_HIGHLIGHTS);
 
   return (
     <Panel header="Pinned Highlights">
       <div className={styles.toggle}>
-        <p className={styles.description}>Pins your last ten highlighted messages above chat</p>
+        <p className={styles.description}>
+          Pins your last{' '}
+          <Input
```

Comment:
i'm not a fan of inlining settings into the description. Please use two setting rows within a Pinned Highlights panel instead.